recipes/pcapplusplus/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class PcapplusplusConan(ConanFile):
+    name = "pcapplusplus"
+    version = "21.05"
+    license = "Unlicense"
+    description = "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets"
+    topics = ("conan", "pcapplusplus", "pcap", "network", "security", "packet")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/seladb/PcapPlusPlus"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "immediate_mode": [True, False],
+    }
+    default_options = {
+        "immediate_mode": False,
+    }
+    generators = "make"
+
+    _source_subfolder = "PcapPlusPlus"
+
+    def configure(self):
+        if self.settings.os not in ["Macos", "Linux"]:
+            raise ConanInvalidConfiguration("%s is not supported" % self.settings.os)
+
+        del self.settings.compiler.libcxx
```

Comment:
libcxx is the used c++ standard library.
Since this is a c++ library, it matters what actual standard library is used.
It should not be removed.

Reply:
The build of the `test_package` doesn't work if I don't remove it. I'm not sure why, is there anything in Conan that I'm missing?

Reply:
The linking error can be found below.
As you can see, the error is about missing c++ symbols: e.g. c++'s string `std::__cxx11::basic_string`.
This means that libcapplusplus did not receive the correct compiler/linker flags or that the build script of pcapplusplus does not propagate the compiler flags correctly.

I fixed it in https://github.com/seladb/conan-center-index/pull/1



<details>

<summary>Linking error of test_package when using clang + libc++</summary>

```
/usr/bin/ld: /home/maarten/.conan/data/pcapplusplus/21.05/_/_/package/6360f9febf03d8a4eacb09559a51ae41f18a7f63/lib/libPacket++.a(SSLHandshake.o): in function `pcpp::SSLClientHelloMessage::ClientHelloTLSFingerprint::toStringAndMD5()':
SSLHandshake.cpp:(.text+0x2a1db): undefined reference to `MD5::operator()(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/usr/bin/ld: /home/maarten/.conan/data/pcapplusplus/21.05/_/_/package/6360f9febf03d8a4eacb09559a51ae41f18a7f63/lib/libPacket++.a(SSLHandshake.o): in function `pcpp::SSLServerHelloMessage::ServerHelloTLSFingerprint::toStringAndMD5()':
SSLHandshake.cpp:(.text+0x2afcb): undefined reference to `MD5::operator()(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/usr/bin/ld: /home/maarten/.conan/data/pcapplusplus/21.05/_/_/package/6360f9febf03d8a4eacb09559a51ae41f18a7f63/lib/libPacket++.a(md5.o): in function `MD5::getHash[abi:cxx11]()':
/home/maarten/.conan/data/pcapplusplus/21.05/_/_/build/6360f9febf03d8a4eacb09559a51ae41f18a7f63/PcapPlusPlus/3rdParty/hash-library/md5.cpp:324: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::reserve(unsigned long)'
/usr/bin/ld: /home/maarten/.conan/data/pcapplusplus/21.05/_/_/package/6360f9febf03d8a4eacb09559a51ae41f18a7f63/lib/libPacket++.a(md5.o): in function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::push_back(char)':
/usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/basic_string.h:1343: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned long, unsigned long, char const*, unsigned long)'
/usr/bin/ld: /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/basic_string.h:1343: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned long, unsigned long, char const*, unsigned long)'
clang-12: error: linker command failed with exit code 1 (use -v to see inv
```

</details>

Reply:
Thanks @madebr for making the required changes! I reviewed your PR and added a few comments